presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java

```diff
@@ -350,6 +356,18 @@ public void addStateChangeListener(StateChangeListener<QueryState> stateChangeLi
         stateMachine.addStateChangeListener(stateChangeListener);
     }
 
+    @Override
+    public Optional<ResourceGroupQueryLimits> getResourceGroupQueryLimits()
+    {
+        return resourceGroupQueryLimits;
+    }
+
+    @Override
+    public void setResourceGroupQueryLimits(ResourceGroupQueryLimits resourceGroupQueryLimits)
+    {
+        this.resourceGroupQueryLimits = Optional.of(requireNonNull(resourceGroupQueryLimits, "resourceGroupQueryLimits is null"));
```

Comment:
After changing this member variable to be an atomic reference, I would just add a sanity check that this is only set one time.  Something like `if (!resourceGroupQueryLimits.compareAndSet(null, resourceGroupLimits)) {` and then throw.

Reply:
@mayankgarg1990 this suggested change caused tests on `TestQueuesDb` to fail - so I made it so we only throw if we attempt to change the limits if they are not empty and different from the current limits